File: src/Model/Statement/PrintStatement.java
package Model.Statement;

import DataTypes.MyIDictionary;
import DataTypes.MyIList;
import Model.Expression.Expression;
import Model.ProgramState;
import Exception.*;
/**
 * Created by vladc on 12.10.2016.
 */
public class PrintStatement implements IStatement {
    private Expression expr;

    public PrintStatement(Expression expr) {
        this.expr = expr;
    }

    @Override
    public String toString() {
        return "Print(" + expr.toString() + ")";
    }

    public ProgramState execute(ProgramState state) throws MyStatementExecutionException {
        MyIList<Integer> out = state.getOut();
        MyIDictionary<String, Integer> symTab = state.getSymTable();

        try {
            out.enqueue(expr.evaluate(symTab));
        }
        catch (ExpressionEvaluationException ex) {
            System.err.println(ex.toString());
            // TODO: 09.11.2016 maybe something else...
        }
        return state;
    }
}

File: Toy-Language-Interpreter/Model/Statement/PrintStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Toy_Language_Interpreter.DataTypes;
using Toy_Language_Interpreter.Model.Expression;
using Toy_Language_Interpreter.MyException;

namespace Toy_Language_Interpreter.Model.Statement
{
    class PrintStatement : IStatement
    {
        private AExpression expr;

        public PrintStatement(AExpression expr)
        {
            this.expr = expr;
        }

        public ProgramState Execute(ProgramState state)
        {
            MyIList<int> outList = state.GetOutList();
            MyIDictionary<string, int> symTab = state.GetSymTab();

            try
            {
                outList.Enqueue(expr.Evaluate(symTab));
            }
            catch (ExpressionEvaluationException e)
            {
                Console.Write(e.ToString());
            }
            return null;
        }

        public override string ToString()
        {
            return "Print(" + expr.ToString() + ")";
        }
    }
}
